cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java

```diff
@@ -458,16 +460,21 @@ private void restartIfNecessary(NodeRef nodeRef, RestartContext restartContext)
 
         restartContext.restartReasons = podNeedsRestart.apply(pod);
 
+        // We try to detect the current roles. If we fail to do so, we optimistically assume the roles did not
+        // change and the desired roles still apply.
+        boolean isBroker = isAlreadyBroker(pod).orElse(nodeRef.broker());
```

Comment:
Would `isCurrentlyBroker`/`isCurrentlyController` sound better?

Reply:
`If we fail to do so` ... I was wondering if this is really possible. From the rest of the code, empty Optional is returned if the label is not present, is that really possible? Someone removing it manually? But wouldn't the operator resets it?

Reply:
@ppatierno I guess the only situation where this would happen is when user somehow manipulates the Pods? But I guess it is better to be safe than sorry and we already have the alternative value here?

Reply:
I agree but in that case it's the `optimistically` part which scared me. If it's a controller instead, our assumption could drive to wrong behaviours. Anyway there is nothing we can do more.